source/validations.ts

```diff
@@ -290,13 +292,29 @@ const validations = {
 	},
 
 	/**
-	 * Checks to see if the instance was created inside of a request handler, which would prevent it from working correctly.
+	 * Checks to see if the instance was created inside of a request handler,
+	 * which would prevent it from working correctly, with the default memory
+	 * store (or any other store with localKeys.)
 	 */
-	creationStack() {
+	creationStack(store: Store) {
 		const { stack } = new Error(
 			'express-rate-limit validation check (set options.validate.creationStack=false to disable)',
 		)
+
 		if (stack?.includes('Layer.handle [as handle_request]')) {
+			if (!store.localKeys) {
+				// This means the user is using an external store, which may be safe.
+				// Print out an error anyways, to alert them of the possibility that
+				// the rate limiter may not work as intended.
+
+				// See the discussion here: https://github.com/express-rate-limit/express-rate-limit/pull/461#discussion_r1626940562.
+				throw new ValidationError(
+					'ERR_ERL_CREATED_IN_REQUEST_HANDLER',
+					'Dynamically creating an express-rate-limit instance for every request may result in unexpected behaviour and unnecessary re-initialization of external stores. Make sure the same data store, prefix, etc. are used.',
```

Comment:
I think we should re-word this a bit, because at the point we log it, we don't know that it's for every request, we just know it was created for _this_ request. (They could be caching instances after the first request.)

Reply:
Maybe just something like this:

```suggestion
					'express-rate-limit instance should *usually* be created at app initialization, not when responding to a request.',
```

... and then send them to the documentation for more information. (It'll automatically get a "See [docs url] for more information." appended to the end.)